### DISCUSSION TITLE
HTTP sender behaviour when non 200 is returned

### DISCUSSION THREAD

**Discussion by muhcdan** [nextgenhealthcare/connect#5884] [Q&A] [answered]
Hi All,

Maintaining a legacy channel. I'm trying to figure out how the response status of the channel gets set to ERROR. Does this happen automatically when a request returns a non 200 response?

I sometimes need to check the 400 errors and I want to sometimes mark the channel status SENT in some cases rather than leaving message in an ERROR status.

**Comment by pacmano1** [comment, marked as answer]
Yes, that is how it works.  If you need to check the response and override that, set the reponse handler to RAW and parse the response code as needed and set the status.  Be mindful you will need to manaully handle all of it.  The parsing of the respone code, the message itself (e.g. JSON.parse if JSON is expected) and setting status.
